src/publicTypes.ts

```diff
@@ -10,80 +10,94 @@ export enum ImgixSourceType {
   WebProxy = 'webProxy',
 }
 
+const ImgixSourceTypeJOI = Joi.string().valid(
+  's3',
+  'gcs',
+  'azure',
+  'webFolder',
+  'webProxy',
+);
+
 type IImgixParamsKey =
   | keyof ImgixUrlParametersSpec['parameters']
   | keyof ImgixUrlParametersSpec['aliases'];
 
-const ImgixParamValueIOTS = t.union(
-  [
-    t.string,
-    t.number,
-    t.boolean,
-    t.undefined,
-    t.null,
-    t.array(t.string),
-    t.array(t.number),
-    t.array(t.boolean),
-  ],
-  'ImgixParamValue',
-);
+const ImgixParamValueJOI = Joi.alternatives()
+  .try(
+    Joi.string(),
+    Joi.number(),
+    Joi.boolean(),
+    Joi.array().items(Joi.string()),
+    Joi.array().items(Joi.number()),
+    Joi.array().items(Joi.boolean()),
+  )
+  .optional()
+  .allow(null);
```

Comment:
For folks unfamiliar with joi could we more details on what this and what it's supposed to do?